common/src/test/java/org/apache/celeborn/common/network/sasl/SaslTestBase.java

```diff
@@ -96,9 +96,7 @@ void authHelper(
           TimeUnit.MILLISECONDS.sleep(10);
         }
       }
-      if (error != null) {
-        throw error;
-      }
+      assertNull(error);
```

Comment:
`assertNull` throws NPE - here we are throwing the `error` `Throwable`.
`testReRegisterationFails`, for example, requires the `error` to be thrown.

Reply:
hmm, I am surprised the tests passed !
+CC @otterc

Reply:
Oh wait, `Throwables.getStackTraceAsString` will serialize the 'cause' as well, heh ...
The intent here though is to throw the actual exception which was raised - not NPE :)